pages/hooks/usereducer.mdx

```diff
@@ -2,6 +2,78 @@
 title: useReducer
 ---
 
+import useReducer from '../../examples/files/hooks/useReducer.js'
+import useReducerRefactor from '../../examples/files/hooks/useReducerRefactor.js'
+import useReducerLazyInit from '../../examples/files/hooks/useReducerLazyInit.js'
+
 # useReducer
 
-Coming soon!
+useReducer is similar to useState since it allows us to control state inside a function component.  
+The difference between the two is that useReducer allows us to handle multiple states and control complex state changes. 
+
+The useReducer hook takes two mandatory arguments: a reducer function and an initial argument. It can also take in an optional third argument called the initializer, which we will discuss later. useReducer returns the current state and a dispatch function.  
+
+`const [state, dispatch] = useReducer(reducer, initialState)`
+
+## Reducer
+
+The reducer function is a pure function that takes in the previous state and an action. It returns a new state. 
+
+```
+const reducer = (prevState, action) => {
+  // do action
+  return newState
+}
+const [state, dispatch] = useReducer(reducer, initialState)
+```
+
+## Dispatch
+
+The dispatch function, when called, passes the action to the reducer.
+
+In the Example code below, the initial state is `No Pet Selected`. When we click the `cat` radio button, dispatch is called with `{type: 'cat'}` as its argument. This argument gets passed to the reducer as `action`. Then the reducer follows the logic `if (action.type === 'cat')` and returns the new state `You selected cat`
+
+<Example code={useReducer}/>
+
+## useState and useReducer
+
+Here we will highlight some of the differences between useState and useReducer. 
```

Comment:
I agree that the Refactor example is a bit long for how much value it adds to the topic.

I think it might be enough to explain the difference without a new example - e.g. we can compare and contrast useReducer + `{ pet, color }` vs. 2 separate useState + `pet` and `color`